packages/babel-parser/src/tokenizer/index.js

```diff
@@ -1215,13 +1215,16 @@ export default class Tokenizer extends ParserErrors {
       hasExponent = true;
       next = this.input.charCodeAt(this.state.pos);
     }
+    let str;
 
     if (next === charCodes.lowercaseN) {
       // disallow floats, legacy octal syntax and non octal decimals
       // new style octal ("0o") is handled in this.readRadixNumber
       if (isFloat || hasLeadingZero) {
         this.raise(start, Errors.InvalidBigIntLiteral);
       }
+      // remove "_" for numeric literal separator
+      str = this.input.slice(start, this.state.pos).replace(/_/g, "");
```

Comment:
I can refactor a bit and avoid duplicating `this.input.slice(start, this.state.pos).replace(/_/g, "");` in several branches.